client/src/components/TreeDropdownField/tests/TreeDropdownField-story.js

```diff
@@ -1,35 +1,50 @@
 import React from 'react';
-// eslint-disable-next-line import/no-extraneous-dependencies
+import { createStore, combineReducers } from 'redux';
+import { Provider } from 'react-redux';
 import { storiesOf } from '@storybook/react';
-// eslint-disable-next-line import/no-extraneous-dependencies
-// import { action } from '@storybook/addon-actions';
-// import { Component as TreeDropdownField } from 'components/TreeDropdownField/TreeDropdownField';
+import { withNotes } from '@storybook/addon-notes';
 import ValueTracker from 'stories/ValueTracker';
 
-// const triggerAction = action('action');
-/*
+import notes from '../README.md';
+import treeDropdownFieldReducer from 'state/treeDropdownField/TreeDropdownFieldReducer';
+import { ConnectedTreeDropdownField as TreeDropdownField } from 'components/TreeDropdownField/TreeDropdownField';
+import mockTree from './mockTree';
+
+// TreeDropdownField depends on having a basic Redux store present to operate
+const store = createStore(
+  combineReducers({ treeDropdownField: treeDropdownFieldReducer }),
+  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
+);
+
+// Returning the mockTree in the fetch mock allows us to manipulate the length of the loading state
+const fetchMock = () => new Promise(resolve => {
+  setTimeout(() => resolve({ json: () => mockTree }), 500);
```

Comment:
only 4 functions? I think we need at least 5 :-)

Reply:
does it need to be 500? does it still function just as well if it's 0?

if it needs to be 500 or more, I'm just worried there may be a bit of a lottery of it working or not depending on if it loads too quickly / too slowly compared to other stuff that gets loaded in

Reply:
I agree, shouldn't this just read `resolve => resolve({ json: Promise.resolve(mockTree) })`?

Reply:
The idea was that seeing the loading state would be useful. There's no risk of it failing, and there's nothing special about `500` - it's just an arbitrary unit of time that seemed slow enough to visualise the loading state but fast enough not to be annoying.

In any case, it's not really that useful, and it's probably better to simulate the loading state with actual state, so I've simplified it per @ScopeyNZ's suggestion. (The nested functions are still required to replicate the behaviour of the Fetch API.)